python/pyspark/ml/param/shared.py

```diff
@@ -583,6 +583,31 @@ def getVarianceCol(self):
         return self.getOrDefault(self.varianceCol)
 
 
+class HasBinary(Params):
+    """
+    Mixin for param binary: If True, all non zero results are set to 1. This is useful for discrete probabilistic models that model binary events rather than integer counts. Default False.
```

Comment:
From toxi.ini we've excluded `cloudpickle.py`,`heapq3.py`, and `shared.py`, we've also done similar exclusions on the Scala side for the auto generated params.